### DISCUSSION TITLE
How to style widgets programatically?

### DISCUSSION THREAD

**Discussion by Kofituo** [gtk-rs/gtk4-rs#1526] [Q&A]
I want to style the widget entirely with Rust code, no CSS. Would that be possible?
Here's what I've tried for a simple button but it doesn't work.

```rust
    // --Snip --

    let button = gtk::Button::with_label("Test button");

    let mut style = button.style_context().color();

    style.set_red(1.0);
    style.set_blue(1.0);
    style.set_green(1.0);
    container.append(&p);
```

The above code has no effect on modifying the color of the button

**Comment by bilelmoussaoui** [comment]
Yes, styling happens through CSS. The style context allows you to retrieve the CSS information in order to manually draw something using those info like the current color.
